wp-includes/components/class-wp-service-worker-navigation-routing-component.php

```diff
@@ -97,19 +97,88 @@ public function serve( WP_Service_Worker_Scripts $scripts ) {
 			 *
 			 * @param string $caching_strategy Caching strategy to use for frontend navigation requests.
 			 */
-			$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_ONLY );
+			$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_FIRST );
+
+			$caching_strategy_args = array(
+				'cache_name' => WP_Service_Worker_Caching_Routes::NAVIGATIONS_CACHE_NAME,
+			);
+			if ( WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_FIRST === $caching_strategy ) {
+				/*
+				 * The value of 2 seconds is informed by the Largest Contentful Paint (LCP) metric, of which Time to
+				 * First Byte (TTFB) is a major component. As long as all assets on a page are cached, then this allows
+				 * for the service worker to serve a previously-cached page and then for LCP to occur before 2.5s and
+				 * so remain within the good threshold.
+				 */
+				$caching_strategy_args['network_timeout_seconds'] = 2;
+			}
+
+			/*
+			 * By default cache only the last 10 pages visited. This may end up being too high as it seems likely that
+			 * most site visitors will view one page and then maybe a couple others.
+			 */
+			$caching_strategy_args['plugins']['expiration']['max_entries'] = 10;
 
 			/**
 			 * Filters the caching strategy args used for frontend navigation requests.
 			 *
 			 * @since 0.2
+			 * @since 0.6 Added $caching_strategy param and initial array has default values provided.
 			 * @see WP_Service_Worker_Caching_Routes::register()
 			 *
-			 * @param array $caching_strategy_args Caching strategy args.
+			 * @param array $caching_strategy_args {
+			 *     Caching strategy args.
+			 *
+			 *     @type string $cache_name              Cache name to store navigation responses.
+			 *     @type int    $network_timeout_seconds Network timeout seconds when NetworkFirst strategy is used.
+			 *     @type array  $plugins                 Configuration for plugins, in particular expiration.
+			 * }
+			 * @param string $caching_strategy Caching strategy being used.
+			 */
+			$caching_strategy_args = apply_filters( 'wp_service_worker_navigation_caching_strategy_args', $caching_strategy_args, $caching_strategy );
```

Comment:
The `wp_service_worker_navigation_caching_strategy` and `wp_service_worker_navigation_caching_strategy_args` filters can be deprecated in favor of the one `wp_service_worker_navigation_caching` filter.